src/main/java/edu/kis/powp/jobs2d/drivers/visitor/DriversCountVisitor.java

```diff
@@ -1,43 +1,46 @@
 package edu.kis.powp.jobs2d.drivers.visitor;
 
-import edu.kis.powp.jobs2d.Job2dDriver;
-import edu.kis.powp.jobs2d.drivers.DriverComposite;
-import edu.kis.powp.jobs2d.drivers.LineSimulatorDriver;
+import edu.kis.powp.jobs2d.drivers.VisitableJob2dDriver;
 import edu.kis.powp.jobs2d.drivers.adapter.DriverDecorator;
+import edu.kis.powp.jobs2d.drivers.composite.IDriverComposite;
 import edu.kis.powp.jobs2d.drivers.logger.LoggerDriver;
 
-import java.util.logging.Logger;
+import java.util.List;
+import java.util.stream.Collectors;
 
-public class DriversCountVisitor implements IVisitor {
+public class DriversCountVisitor implements IDriverVisitor {
     private long count = 0;
 
-    public long getDriversCount(){
-        return count;
+    public long getDriversCount() {
+        long tmpCount = count;
+        count = 0;
+        return tmpCount;
```

Comment:
OK, but this is not a stateless approach. In stateless approach each accept call is totally independent and generates the appropriate result to the visited node.